user/User.cc

```diff
@@ -1,135 +1,138 @@
 #include "User.h"
+#include "../utils/StatusPrinter.h"
+#include "../devices/AirConditioner.h"
+#include "../devices/Heater.h"
+#include "../devices/Light.h"
 
 namespace SmartHome {
-    User::User(int numRooms, int numSensorsPerRoom) : userHouse(numRooms, numSensorsPerRoom) {
-        devices.push_back(new Device(LIGHT));
-        devices.push_back(new Device(HEATER));
+     User::User(int numRooms) : userHouse(numRooms) {
+        devices.push_back(new Light());
+        devices.push_back(new Heater());
+        devices.push_back(new AirConditioner());
     }
 
-    User::User(const User& other) : userHouse(other.userHouse) {
-        devices.reserve(other.devices.size());
-        for (Device* device : other.devices) {
-            devices.push_back(new Device(*device));
+    User::~User() {
+        for (auto device : devices) {
+            delete device;
         }
     }
 
-    User::User(User&& other) : userHouse(std::move(other.userHouse)), devices(std::move(other.devices)) {
-        other.devices.clear();
+    User::User(const User& other) : userHouse(other.userHouse) {
+        copyDevices(other);
     }
 
-    User& User::operator=(User&& other) {
-        if (this != &other) {
-            userHouse = std::move(other.userHouse);
+    User::User(User&& other) noexcept : userHouse(std::move(other.userHouse)) {
+        moveDevices(other);
+    }
 
-            for (Device* device : devices) {
+    User& User::operator=(const User& other) {
+        if (this != &other) {
+            for (auto device : devices) {
                 delete device;
             }
-            devices = std::move(other.devices);
-            other.devices.clear();
+            devices.clear();
+
+            userHouse = other.userHouse;
+
+            copyDevices(other);
         }
         return *this;
     }
 
-    User& User::operator=(const User& other) {
+    User& User::operator=(User&& other) noexcept {
         if (this != &other) {
-            userHouse = other.userHouse;
-
-            for (Device* device : devices) {
+            for (auto device : devices) {
                 delete device;
             }
             devices.clear();
 
-            devices.reserve(other.devices.size());
-            for (Device* device : other.devices) {
-                devices.push_back(new Device(*device));
-            }
+            userHouse = std::move(other.userHouse);
+
+            moveDevices(other);
         }
         return *this;
     }
 
-    User::~User() {
-        for (Device *device: devices) {
-            delete device;
+
+    void User::copyDevices(const User& other) {
+        for (auto device : other.devices) {
+            devices.push_back(device->clone());
         }
     }
 
+    // Utility function for clearing and moving devices
+    void User::moveDevices(User& other) {
+        devices = std::move(other.devices);
+        other.devices.clear();
+    }

```

Comment:
not 100% this works as expected but we'll double check in the future presentations

Reply:
da, am observat acum ca nu e tocmai the best way. voi corecta si deschid PR, am observat si ca namespace-urile nu imi respecta naming conventions si le voi schimba in lowercases